courses/management/commands/check_program_requirements.py

```diff
@@ -0,0 +1,87 @@
+"""
+Checks programs for a valid requirements tree. A valid tree is one that:
+a) exists
+b) has all the courses in the program accounted for as either an elective
+   or a required course
+
+This won't fix the issue for you - do that via Django Admin - but it will tell
+you if there are any.
+"""
+
+from django.core.management import BaseCommand
+from django.db.models import Q
+
+from courses.api import check_program_for_orphans
+from courses.models import Program
+
+
+class Command(BaseCommand):
+    """
+    Upgrade enrollment for legacy paid learners
+    """
+
+    help = "Checks program(s) for valid requirements trees"
+
+    def add_arguments(self, parser) -> None:
+        parser.add_argument(
+            "--program",
+            action="append",
+            help="Program to check.",
+            nargs="*",
+        )
+
+        parser.add_argument(
+            "--live", action="store_true", help="Check only live programs."
+        )
+
+    def handle(self, *args, **kwargs):  # pylint: disable=unused-argument
+        if kwargs["program"] is not None and len(kwargs["program"]) > 0:
+            numeric_ids = []
+            readable_ids = []
+
+            [
+                readable_ids.append(id[0])
+                if not id[0].isnumeric()
+                else numeric_ids.append(id[0])
+                for id in kwargs["program"]
+            ]
+
+            programs_qset = Program.objects.filter(
+                Q(id__in=numeric_ids) | Q(readable_id__in=readable_ids)
+            )
+        else:
+            programs_qset = Program.objects
+
+        if kwargs["live"]:
+            programs_qset = programs_qset.filter(live=True)
+
+        for program in programs_qset.all():
+            orphans = check_program_for_orphans(program, False)
+
+            if len(orphans) > 0:
+                if program.get_requirements_root() is None:
+                    self.stdout.write(
+                        self.style.ERROR(
+                            f"Program {program.readable_id} has no requirements tree!"
```

Comment:
I'm not sure if it's possible, but is there a way to capture the error output when you call `check_program_for_orphans` so you don't need to interpret the list of courses returned and re-output the error?

Reply:
Worked this out in https://github.com/mitodl/mitxonline/pull/1244/commits/048bf39107e0b7cd957f849856917cbe2e0430f0 but with the caveat that it turns off log message suppression in the command (so running it will potentially generate Sentry messages). I'm on the fence about whether or not this is good or bad.

Reply:
@jkachel You might be able to add another error type and check for it here: https://github.com/mitodl/mitxonline/blob/e555739257d05c18ece336ae5567ab581d309553/main/sentry.py#L26.